File: 0371-sum-of-two-integers/0371-sum-of-two-integers.cpp
class Solution {
public:
    int getSum(int a, int b) {
        int carry = 0;
        int mask = 0;
        for(int p = 0; p < 32; p++){
            int x, y;
            x = (a & (1 << p));
            y = (b & (1 << p));
            if(!x && !y){
                if(carry){
                    carry = 0;
                    mask |= (1 << p);
                }
            }
            else if(x && y){
                if(carry) mask |= (1 << p);
                else carry = 1;
            }
            else{
                if(!carry){
                    mask |= (1 << p);
                }
            }
        }
        return mask;
    }
};

File: 0371-sum-of-two-integers/0371-sum-of-two-integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = (1 << 32) - 1
        while (mask & b) > 0:
            s = (a ^ b)
            carry = (a & b) << 1
            a, b = s, carry
            # a, b = (a ^ b), (a & b) << 1
        return (a & mask) if b > 0 else a

            